src/integTest/groovy/net/wasdev/wlp/gradle/plugins/TestSpringBootApplication.groovy

```diff
@@ -64,7 +64,7 @@ public class TestSpringBootApplication extends AbstractIntegrationTest{
 
 
     @Test
-    public void test_spring_boot_apps() {
+    public void test_spring_boot_apps_20() {
         try {
             addApplToServerXml("src/main/liberty/config/server.xml",
```

Comment:
Should you should setting the `buildFilename` to the `test_spring_boot_apps_20.gradle` here? I think runTasks() is using `springboot_archive.gradle` up on line 24.

Reply:
Setting the build.gradle file for each test case might fix the problem you're having later with the other tests when you try to add the app to the server.xml.

Instead of writing to the server.xml file you could also add  different server.xml files for the other builds. This way you wouldn't need to edit them during the tests. Along with the server.xml files, you'd just need to add this to the test build.gradle:
```
liberty {
    server {
        configFile = file('src/main/liberty/config/thisTestServer.xml')
    }    
}
```